File: objectiveai/objectiveai/src/functions/executions/response/unary/object.rs
use crate::functions::executions::response;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
pub enum Object {
    #[serde(rename = "scalar.function.execution")]
    ScalarFunctionResponse,
    #[serde(rename = "vector.function.execution")]
    VectorFunctionResponse,
}

impl From<response::streaming::Object> for Object {
    fn from(value: response::streaming::Object) -> Self {
        match value {
            response::streaming::Object::ScalarFunctionResponseChunk => {
                Object::ScalarFunctionResponse
            }
            response::streaming::Object::VectorFunctionResponseChunk => {
                Object::VectorFunctionResponse
            }
        }
    }
}

File: objectiveai-js/src/functions/executions/response/unary/object.ts
import { z } from "zod";

export const FunctionsExecutionsResponseUnaryObjectSchema = z.enum(["scalar.function.execution","vector.function.execution"]).meta({ title: "functions.executions.response.unary.Object" });
export type FunctionsExecutionsResponseUnaryObject = z.infer<typeof FunctionsExecutionsResponseUnaryObjectSchema>;
